src/main/java/com/cognifide/aemrules/checks/ContentResourceShouldBeNullCheckedCheck.java

```diff
@@ -0,0 +1,266 @@
+        insideIfStatement = true;
+
+        if (isThisAResourceNullCheck(tree, NOT_EQUAL)) {
+            updateNullCheckedResources(tree, true);
+        } else if (isThisAResourceNullCheck(tree, EQUAL)) {
+            insideEqualNullCheck = true;
+        }
+        super.visitIfStatement(tree);
+        if (returnOccurredInsideEqualNullCheck) {
+            updateNullCheckedResources(tree, true);
+        }
+        cleanFlagsAfterIfStatement(tree);
+    }
+
+    @Override
+    public void visitAssignmentExpression(AssignmentExpressionTree tree) {
+        if (isGetContentResourceUsedOnResource(tree)) {
+            contentResources.put(tree.variable().firstToken().text(), false);
+        }
+        super.visitAssignmentExpression(tree);
+    }
+
+    @Override
+    public void visitMethodInvocation(MethodInvocationTree tree) {
+        externalLibraryNullChecks(tree);
+        equalsNullCheck(tree);
+        chainedMethodsNullCheck(tree);
+        super.visitMethodInvocation(tree);
+    }
+
+    @Override
+    public void visitVariable(VariableTree tree) {
+        if (isResourceInitialized(tree)) {
+            shouldBeNullChecked(tree);
+        }
+        super.visitVariable(tree);
+    }
+
+    @Override
+    public void visitClass(ClassTree tree) {
+        contentResources.clear();
+        super.visitClass(tree);
+    }
+
+    @Override
+    public void visitMethod(MethodTree tree) {
+        super.visitMethod(tree);
+        returnOccurredInsideEqualNullCheck = false;
+    }
+
+    @Override
+    public void visitReturnStatement(ReturnStatementTree tree) {
+        if (insideEqualNullCheck) {
+            returnOccurredInsideEqualNullCheck = true;
+        }
+        super.visitReturnStatement(tree);
+    }
+
+    private void shouldBeNullChecked(VariableTree tree) {
+        if (isGetContentResourceUsedOnPage(tree)) {
+            contentResources.put(tree.simpleName().name(), false);
+        }
+    }
+
+    private void cleanFlagsAfterIfStatement(IfStatementTree tree) {
+        insideIfStatement = false;
+        updateNullCheckedResources(tree, false);
+        insideEqualNullCheck = false;
+    }
+
+    private void equalsNullCheck(MethodInvocationTree tree) {
+        if (!contentResources.getOrDefault(tree.firstToken().text(), true) &&
+            !returnOccurredInsideEqualNullCheck
+        ) {
+            context.reportIssue(this, tree, RULE_MESSAGE);
+        }
+    }
+
+    private void updateNullCheckedResources(IfStatementTree tree, boolean value) {
+        if (isRightSideNullCheck(tree)) {
+            contentResources.replace(tree.condition().firstToken().text(), value);
+        } else {
+            contentResources.replace(tree.condition().lastToken().text(), value);
+        }
+    }
+
+    private void externalLibraryNullChecks(MethodInvocationTree tree) {
+        if (isNonNullUsed(tree)) {
+            contentResources.replace(tree.arguments().get(0).toString(), true);
```

Comment:
0 is a magic number please provide constant for this

Reply:
done